app_dart/lib/src/request_handlers/scheduler/request_subscription.dart

```diff
@@ -47,7 +47,12 @@ class SchedulerRequestSubscription extends SubscriptionHandler {
   Future<Body> post() async {
     BatchRequest request;
     try {
-      final String rawJson = String.fromCharCodes(base64Decode(message.data!));
+      String rawJson = String.fromCharCodes(base64Decode(message.data!));
+      try {
+        rawJson = String.fromCharCodes(base64Decode(message.data!));
+      } on FormatException {
+        rawJson = message.data!;
+      }
       log.info('rawJson: $rawJson');
       final Map<String, dynamic> json = jsonDecode(rawJson) as Map<String, dynamic>;
       request = BatchRequest.fromJson(json);
```

Comment:
Can we separate the logic into two parts?

```
final String data = message.data!;
Map<String, dynamic> jsonData;
log.info('rawJson: $data');
try {
      jsonData = jsonDecode(data) as Map<String, dynamic>;
    } on FormatException {
      jsonData = json.decode(String.fromCharCodes(base64.decode(data))) as Map<String, dynamic>;
    }

try {
      request = BatchRequest.fromJson(jsonData);
    } catch (e) {
      log.severe('Failed to construct BatchRequest from message');
      log.severe(e);
      throw BadRequestException(e.toString());
    }
```

Reply:
done